src/Utils/BeanDescriptor.php

```diff
@@ -607,6 +607,7 @@ public function generatePhpCode(): ?FileGenerator
             $class->setExtendedClass(AbstractTDBMObject::class);
             $file->setUse(AbstractTDBMObject::class);
         } else {
+            /** @var class-string $extends */
```

Comment:
Cannot this be type-hinted in `BeanDescriptorInterface::getExtendedBeanClassName()` (and `BeanDescriptor::getExtendedBeanClassName()` ?

Reply:
I don't think so, since those methods return the class names without their associated namespace

Reply:
Okay, so this is a "`class-string`" because we do have the `use` statements generated with:
```
$classes = $this->generateExtendsAndUseStatements($parentFk);
```

Maybe a more robust way would be to concatenate the `$this->beanNamespace` and `assert`. We could also move the `use` to this specific block and remove it from `BeanDescriptor::generateExtendsAndUseStatements()` (and rename) ?